### ISSUE TITLE
Segmentation fault in/near yaml_duplication_checker.rb

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15460]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Steps to reproduce the problem

Happened when pressing Ctrl+C while running. Not sure what caused it or if it can be reproduced, but ask if the issue is not immediately self-evident. Forward to respective tracker if core cause is not Rubocop.

### Expected behavior

No crash

### Actual behavior

```
/opt/homebrew/Cellar/ruby/4.0.5/lib/ruby/4.0.0/arm64-darwin25/enc/utf_16be.bundle: warning: failed to load encoding (UTF-16BE); use ASCII-8BIT instead
/opt/homebrew/Cellar/ruby/4.0.5/lib/ruby/4.0.0/arm64-darwin25/enc/utf_16be.bundle: [BUG] vm_call_cfunc: cfp consistency error (0x000000075091bc48, 0x000000075091bc10)
ruby 4.0.5 (2026-05-20 revision 64336ffd0e) +PRISM [arm64-darwin25]

-- Control frame information -----------------------------------------------
c:0019 p:---- s:0126 e:000125 l:y b:0001 TOP    [FINISH]
c:0018 p:---- s:0123 e:000122 l:y b:---- CFUNC  :_native_parse
c:0017 p:0028 s:0116 e:000115 l:y b:0001 METHOD /opt/homebrew/lib/ruby/gems/4.0.0/gems/psych-5.4.0/lib/psych/parser.rb:62
c:0016 p:0037 s:0110 e:000109 l:y b:0001 METHOD ./rubocop/lib/rubocop/yaml_duplication_checker.rb:10
c:0015 p:0017 s:0101 E:0007d8 l:y b:0001 METHOD ./rubocop/lib/rubocop/config_loader.rb:234
c:0014 p:0027 s:0094 e:000093 l:y b:0001 METHOD ./rubocop/lib/rubocop/config_loader.rb:81
c:0013 p:0012 s:0085 e:000084 l:y b:0001 METHOD ./rubocop/lib/rubocop/config_loader.rb:56
c:0012 p:0022 s:0072 e:000071 l:y b:0001 METHOD ./rubocop/lib/rubocop/config_loader.rb:128
c:0011 p:0089 s:0064 e:000060 l:y b:0001 METHOD ./rubocop/lib/rubocop/config_store.rb:73
c:0010 p:0007 s:0055 e:000054 l:y b:0001 METHOD ./rubocop/lib/rubocop/config_store.rb:52
c:0009 p:0010 s:0051 e:000050 l:n b:---- BLOCK  ./rubocop/lib/rubocop/result_cache.rb:92
c:0008 p:0015 s:0048 e:000047 l:y b:0001 METHOD ./rubocop/lib/rubocop/cache_config.rb:9
c:0007 p:0023 s:0043 e:000042 l:y b:0001 METHOD ./rubocop/lib/rubocop/result_cache.rb:91
c:0006 p:0050 s:0036 e:000034 l:y b:0001 METHOD ./rubocop/lib/rubocop/cli.rb:172
c:0005 p:0018 s:0030 e:000029 l:n b:---- BLOCK  ./rubocop/lib/rubocop/cli.rb:49
c:0004 p:0011 s:0027 e:000026 l:y b:0001 METHOD ./rubocop/lib/rubocop/cli.rb:89
c:0003 p:0069 s:0017 e:000016 l:y b:0001 METHOD ./rubocop/lib/rubocop/cli.rb:45
c:0002 p:0107 s:0008 E:002260 l:n b:---- EVAL   ./rubocop:15 [FINISH]
c:0001 p:0000 s:0003 E:0024e0 l:y b:---- DUMMY  [FINISH]

```

### RuboCop configuration

```yaml
/dev/null
```

### RuboCop version

```shell
System Version: macOS 26.5.1 (25F80)
Model Identifier: MacBookAir10,1
ruby 4.0.5 (2026-05-20 revision 64336ffd0e) +PRISM [arm64-darwin25]
rubocop     2026-07-14 84bcf77f7
```

**Comment by koic** [comment]
This is a crash in the Ruby interpreter itself, not in RuboCop. A `[BUG] vm_call_cfunc: cfp consistency error` report can only be produced by a defect in CRuby or a C extension; RuboCop's code at the top of the stack is an ordinary use of `Psych::Parser`.

The mechanism matches a known CRuby bug: when a signal (your Ctrl+C) arrives while Ruby is lazily loading an encoding (note the `failed to load encoding (UTF-16BE)` warning right before the crash, triggered from Psych's native parser), the
  interrupt can longjmp over the code that pops the control frame, leaving the VM in an inconsistent state.

This is already tracked upstream as https://bugs.ruby-lang.org/issues/21967, which is still open and affects Ruby 2.7.8 through 4.0.5. Closing as this is not actionable in RuboCop.
